### PR TITLE
Crypto crash fix

### PR DESCRIPTION
**Pull request by wxsBSD**
Fix a crash in OpenSSL due to not properly initializing with respect to threading.

### CODE REVIEW

**Comment by wxsBSD** [comment]
I'll shoot you a mail with exact hashes and signatures that reproduce this. My laptop is currently burning through another round of testing but I am fairly confident this fixes the problem.

File: libyara/libyara.c [hunk 1]
```diff
@@ -112,6 +143,13 @@ YR_API int yr_finalize(void)
 {
   yr_re_finalize_thread();
 
+  #ifdef HAVE_LIBCRYPTO
```
Comment (plusvic) [inline libyara/libyara.c]: This should appear after the init_count > 0 check

Comment (wxsBSD) [inline libyara/libyara.c]: Fixed! It was a bad merge on my part. Thanks for catching it! :)

File: libyara/libyara.c [hunk 2]
```diff
@@ -115,6 +145,13 @@ YR_API int yr_finalize(void)
   if (--init_count > 0)
     return ERROR_SUCCESS;
 
+  #ifdef HAVE_LIBCRYPTO
+  int i;
```
Comment (plusvic) [inline libyara/libyara.c]: Just a small note: I recently started to comply with the C89 standard. Declaring "i" in the middle of the function is not accepted by C89, the declaration should be moved to the beginning of the function.

**Comment by wxsBSD** [comment]
I've got it fixed locally. Will push whenever I'm back on stable wifi, which may be as late as Monday. I'll try to keep this in mind while I work on my next PR. 😛

**Pull request merged**